crates/state/src/client_state.rs

```diff
@@ -44,9 +44,15 @@ pub struct ClientState {
     /// Height at which we'll create the L2 genesis block from.
     pub(super) genesis_l1_height: u64,
 
+    /// The depth at which we accept blocks to be finalized.
+    pub(super) finalization_depth: u64,
+
     /// Hash of verification state at `genesis_l1_height`. The hash is computed via
     /// [`super::l1::HeaderVerificationState::compute_hash`]
     pub(super) genesis_l1_verification_state_hash: Option<Buf32>,
+
+    /// Internal states according to each block height.
+    pub(crate) int_states: BTreeMap<u64, InternalState>,
```

Comment:
Looks like  `StateQueue` can be used here to deal with `PartialEq` issue that you mentioned earlier.

Reply:
Yeah it'd probably be fine to use that actually.

Reply:
Did.